client/state/jetpack/credentials/test/reducer.js

```diff
@@ -1,61 +1,123 @@
 import { markCredentialsAsInvalid, markCredentialsAsValid, testCredentials } from '../actions';
-import { testRequestStatus } from '../reducer';
+import { testRequestStatus, testStatus } from '../reducer';
 
 describe( 'reducer', () => {
 	describe( 'testRequestStatus', () => {
 		test( 'should return the initial state', () => {
 			expect( testRequestStatus( undefined, {} ) ).toEqual( {} );
 		} );
 
-		test( 'status should be valid if the credentials are okay', () => {
+		test( 'should return true if we are requesting to test credentials', () => {
 			const stateIn = {};
-			const action = markCredentialsAsValid( 123000, 'main' );
+			const action = testCredentials( 123000, 'main' );
+			const stateOut = testRequestStatus( stateIn, action );
+
+			expect( stateOut ).toEqual( {
+				123000: {
+					main: true,
+				},
+			} );
+		} );
 
+		test( 'should return false if credentials are valid', () => {
+			const stateIn = {};
+			const action = markCredentialsAsValid( 123000, 'main' );
 			const stateOut = testRequestStatus( stateIn, action );
+
 			expect( stateOut ).toEqual( {
 				123000: {
-					main: 'valid',
+					main: false,
 				},
 			} );
 		} );
 
-		test( 'status should be invalid if the credentials are not okay', () => {
+		test( 'should return false if credentials are invalid', () => {
 			const stateIn = {};
 			const action = markCredentialsAsInvalid( 234000, 'main' );
+			const stateOut = testRequestStatus( stateIn, action );
 
+			expect( stateOut ).toEqual( {
+				234000: {
+					main: false,
+				},
+			} );
+		} );
+
+		test( 'should return true if credentials are valid (or invalid) and we request a new credentials test for same role', () => {
+			const stateIn = {
+				234000: {
+					main: false,
+				},
+			};
+			const action = testCredentials( 234000, 'main' );
 			const stateOut = testRequestStatus( stateIn, action );
+
 			expect( stateOut ).toEqual( {
 				234000: {
-					main: 'invalid',
+					main: true,
+				},
+			} );
+		} );
+
+		test( 'should keep existing roles request statuses after requesting a new credentials test for a new role', () => {
```

Comment:
Do we also need a test to ensure that we store data for multiple sites at the same time, and data of a new site doesn't overwrite data of previously existing site?

Reply:
Yes, great suggestion. Just added one test case for that!